### PR TITLE
feat: Site dashboard & Soil info sheet displays info resulting from Soil ID API

### PR DESCRIPTION
**Pull request by knipec**
## Description
(Creating from Ruxandra's work)
Use Soil ID API to get soil id matches on the site location screen, and access them within a given site via redux to display them on soil id screen and soil id soil info screen

### Related Issues
Implements [#653](https://github.com/techmatters/terraso-product/issues/653)

### Verification steps
Look at the soil ID info screen for a given soil (need the corresponding client-shared work)

### CODE REVIEW

File: dev-client/package.json
Comment (knipec) [inline dev-client/package.json:64]: TODO: Will need to update this before merging!

**Review by paulschreiber — COMMENTED** [review]
LGTM, but others should review/approve.

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (knipec) [inline dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:142]: Question: I believe this should be fine as-is, but do you think we should get in the habit of preferring `useSoilIdData` instead of some places using the selector directly? Just feels like doing this requires more thought about when we'd need to request potentially new data?

File: dev-client/src/screens/LocationScreens/components/soilId/SoilIdMatchesSection.tsx
Comment (knipec) [inline dev-client/src/screens/LocationScreens/components/soilId/SoilIdMatchesSection.tsx:43]: Same question as in `LocationDashboardContent` about preferring `useSoilIdData`

**Review by knipec — COMMENTED** [review]
Lol finally reviewing the thing I technically put out for PR

File: dev-client/src/model/soilId/soilIdRanking.ts
Comment (knipec) [inline dev-client/src/model/soilId/soilIdRanking.ts:34]: Issue: This is backwards. The item with the smallest rank should be returned (it might always be the item with rank 0 but I don't know for usre)

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (paulschreiber) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:238]: @CourtneyLee333 What's the fallback string here?

Comment (knipec) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:238]: Also, @paulschreiber or @shrouxm I'm assuming we never expect a time when there is a top soil match but it is missing the name attribute, right?

Comment (paulschreiber) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:238]: No, but we should still display something sensible to the user in that case.

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (tm-ruxandra) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:142]: I can give this some more thought - I agree with your point that this potentially requires more reasoning about when data gets loaded.

File: dev-client/src/model/soilId/soilIdRanking.ts
Comment (tm-ruxandra) [reply dev-client/src/model/soilId/soilIdRanking.ts:34]: Good catch!

Comment (tm-ruxandra) [reply dev-client/src/model/soilId/soilIdRanking.ts:34]: You're right - I was on brain-autopilot and forgot that JS' reduce method does this automatically.

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (tm-ruxandra) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:238]: Will discuss with design to get clarification on copy text to show here in the empty case.

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (tm-ruxandra) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:142]: Will refactor to use this.

File: dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx
Comment (tm-ruxandra) [reply dev-client/src/screens/LocationScreens/LocationDashboardContent.tsx:238]: From Courtney: "None available" is the required copy text here.

File: dev-client/src/hooks/soilIdHooks.ts
Comment (paulschreiber) [inline dev-client/src/hooks/soilIdHooks.ts:69]: question: should this be `&&` instead of `||`?

**Review by paulschreiber — APPROVED** [review]

File: dev-client/src/hooks/soilIdHooks.ts
Comment (tm-ruxandra) [reply dev-client/src/hooks/soilIdHooks.ts:69]: No. If either parameter is different from what was most recently loaded, Redux should fetch new matches.

File: dev-client/__tests__/integration/SlopeScreen-test.tsx
Comment (tm-ruxandra) [reply dev-client/__tests__/integration/SlopeScreen-test.tsx:1]: It's because the state is part of the SoilID Redux slice, which this snapshot uses because it's touching soil data (which is part of the slice).

**Pull request merged**
